### ISSUE TITLE
Replace repo Hylafax with HylaFAX+ built from source to add support for color grayscale JPEG fax

### ISSUE THREAD

**Issue by chris001** [joshnorth/UbuntuPIAF#1]
I see in the comments on the install shell script, that you reverted from the Hylafax+ source build back to the repo version of plain Hylafax due to difficulties you had in doing to build from source?

Would it be possible for you to take another look and replace the Hylafax pulled in from the repo, with Hylafax+ built from source.

As you may know Hylafax+ is much better to be running because it automatically adds color and grayscale support (via JPEG and JBIG) on both send and receive.

Latest version of HylaFax+ is 5.5.6 from Sourceforge

If you need help I've built Hylafax+ from source and it compiled successfully.

What exactly was the difficulty you ran into while trying to get the Hylafax+ build to work for this ubuntu 14 install script?

**Comment by joshnorth** [comment]
I'll try it again, it'll be a few days though.  I had some issues with GCC
and dependencies on different architectures previously - since the script
was intended to run on rPi's, beagle bones, OpenVZ images, etc.
IIRC there was something about using gcc/gold linker but I forget - had
another kid since then.

**Comment by chris001** [comment]
Congrats on having another kid! (cigar)
It's possible some dependencies are named different names on different platforms, so that's why some builds failed due to missing dependencies.
For fun I tried making a deb on 32 bit ubuntu and it completed and installed but I've not idea if it's truly valid and working because I'd previously installed from source hylafax+ 5.5.6 over the existing hylafax 6.0.6 and now the "custom destination hylafax" rings forever.
iPBX is working to the extent that it's able to receive incoming and make outgoing voice calls.
A voice call goes to an extension, ivr, voicemail, or otherwise is handled fine.
A fax call, iPbx seems to also be auto-detecting the incoming fax tones, because it treats those incoming calls differently, no ivr, no extension, they "ring forever". You can hear that when you listen in to the call on the fax machine that is trying to send the fax to ipbx.
I assume what's happening is asterisk is trying to send the call to the "custom destination: hylafax" and some process, iaxmodem, vgetty, faxgetty, hfaxd, or faxq, is failing to take the call, or failing to receive notification that it has a call to take, somewhere along the way.
Is there a diagram or documentation that shows how/where iaxmodem/faxgetty/vgetty/hfaxd/faxq are plugged together, their baseline required config files, what are the expected log file entries under various inputs, etc...?
In the interest of troubleshooting this because it just might be a little thing that needs tweaking and it'll be answering and receiving faxes again.

**Comment by joshnorth** [comment]
Yes, a lot of that was just me trying, reverting snapshot, trying again. I think I remember the IAX modem processes didn't want to talk to the hylafax daemon either. I'll try to draw it out tonight and build on a test machine.

**Comment by chris001** [comment]
Here's the complete build spec for the rpm binary package.
https://apps.fedoraproject.org/packages/hylafax+/sources/spec
It's very human readable and the information contained in it could be reused for building a deb binary, and/or for building from source via shell script successfully.  
It shows essential bits like, all packages required to build without issue, and all packages required to install the binary in its runtime environment without issue.  The stuff that was giving you grief in other words. Bonus, it was made fully or in large part, by Lee Howard, the author/maintainer of hylafax+. Very cool.
